gpii/node_modules/testing/src/RunTestDefs.js

```diff
@@ -0,0 +1,240 @@
+    };
+};
+
+gpii.test.testDefToServerEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.serverEnvironment", "gpii.test.standardServerSequenceGrade");
+};
+
+gpii.test.testDefToCouchEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.couchEnvironment", "gpii.test.couchSequenceGrade");
+};
+
+gpii.test.testDefToSnapshotEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.couchEnvironment", "gpii.test.snapshotSequenceGrade");
+};
+
+gpii.test.testDefToSnapshotWithInitialStateEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.couchEnvironment", "gpii.test.initialStateSequenceGrade");
+};
+
+/*
+
+
+gpii.test.testDefToServerEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.serverEnvironment", testDef.sequenceGrade || "gpii.test.standardServerSequenceGrade");
+};
+
+gpii.test.testDefToCouchEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.couchEnvironment", testDef.sequenceGrade || "gpii.test.couchSequenceGrade");
+};
+
+gpii.test.testDefToSnapshotEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.couchEnvironment", testDef.sequenceGrade || "gpii.test.snapshotSequenceGrade");
+};
+
+gpii.test.testDefToSnapshotWithInitialStateEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.couchEnvironment", testDef.sequenceGrade || "gpii.test.initialStateSequenceGrade");
+};
+
+ */
+
+
+/* Build a test fixture for integration/acceptance tests operating the stereotypical workflow -
+ * snapshot, login, expectConfigured, logout and expectRestored
+ */
+gpii.test.buildSingleTestFixture = function (testDef, rootGrades) {
+    testDef.gradeNames = fluid.makeArray(testDef.gradeNames).concat(fluid.makeArray(rootGrades));
+
+    // TODO: Discuss whether this is useful/advisable.
+    testDef.expect = testDef.initialState ? 6 : 5;
+
+    testDef.sequenceGrade = testDef.initialState ? "gpii.test.initialStateSequenceGrade" : "gpii.test.snapshotSequenceGrade";
+
+    return testDef;
+};
+
+/** Build a fixture set suitable for sending to kettle.test.bootstrapServer that includes a non-flat
+ * array of sequences within the member `sequenceSegments` as well as a base record to be merged into each testDef
+ * @param {Array} fixtures - An array of proto-testDefs, each members `name` and `expect` with the same semantic
+ * as a standard testDef, and a member `sequenceSegments` whose elements may themselves be arrays of sequences.
+ * @param {Object} baseTestDef - A base test def record, holding members `configName`, `configPath` and other
+ * members appropriate for this options structure.
+ */
+// TODO: Review that we shouldn't use gradeNames and unify with buildSingleTestFixture's "rootGrades" pattern
```

Comment:
What do you mean by "we shouldn't use gradeNames"?

Reply:
No idea, I didn't write that, I just moved it.  [Here it is in master](https://github.com/GPII/universal/blob/master/gpii/node_modules/testing/src/Testing.js#L453).

Reply:
Happy to bin it if it doesn't seem like a useful note for the future.

Reply:
Yes, please bin - we can consider whatever intent was behind the comment is enacted moving to sequence grades

Reply:
Done, will commit once I fix the broken prefs server tests.